### PR TITLE
add com.dhsdevelopments.Climaxima

### PR DESCRIPTION
**Pull request by lokedhs**
Climaxima is a user interface built on top of Maxima, a CAS based
on Macsyma which was started in the 1960's.

The application has the ability to perform regular algebra functions,
do integration and differentiation, solve differential equations and
a lot of other things.

The Climaxima interface adds proper mathematical rendering of
equations as well as other features to  make using Maxima much
easier.

Climaxima is still a work in process, but it has reached a point where
it can be useful for more people.

The project is hosted here: https://github.com/lokedhs/maxima-client

For more information about Maxima: http://maxima.sourceforge.net/

### CODE REVIEW

File: com.dhsdevelopments.Climaxima.json [hunk 1]
```diff
@@ -0,0 +1,170 @@
+{
+    "app-id": "com.dhsdevelopments.Climaxima",
+    "runtime": "org.gnome.Platform",
+    "runtime-version": "3.30",
```
Comment (nedrichards) [inline com.dhsdevelopments.Climaxima.json]: is there a reason 3.32 doesn't work here?

File: com.dhsdevelopments.Climaxima.appdata.xml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (nedrichards) [inline com.dhsdevelopments.Climaxima.appdata.xml]: This won't pass the appdata requirements https://github.com/flathub/flathub/wiki/App-Requirements#Appstream the guidelines page might help too https://github.com/flathub/flathub/wiki/AppData-Guidelines

File: com.dhsdevelopments.Climaxima.json [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+            "build-commands": [
+                "mkdir -p /app/share/metainfo",
+                "cp com.dhsdevelopments.Climaxima.appdata.xml /app/share/metainfo/com.dhsdevelopments.Climaxima.appdata.xml",
+                "mkdir -p /app/share/applications",
+                "cp com.dhsdevelopments.Climaxima.desktop /app/share/applications/com.dhsdevelopments.Climaxima.desktop"
+            ]
+        }
+    ],
+    "finish-args": [
+        "--socket=x11",
```
Comment (nedrichards) [inline com.dhsdevelopments.Climaxima.json]: if you use --socket=x11 then you should add --share=ipc as well http://docs.flatpak.org/en/latest/sandbox-permissions.html

**Comment by nedrichards** [comment]
bot, build com.dhsdevelopments.Climaxima

**Comment by nedrichards** [comment]
@lokedhs btw. you can use the bot, build command in this thread to test a build once you push a new commit as well.

**Comment by nedrichards** [comment]
As you are upstream we normally really want as many of the desktop-integration and build steps to go there as possible since they benefit everyone in the Linux ecosystem that way, also it's a more natural workflow for developers that way.

File: com.dhsdevelopments.Climaxima.json (hunk 1 above)
Comment (lokedhs) [reply com.dhsdevelopments.Climaxima.json]: No reason. In fact, it works with freedesktop, so I have changed this.

File: com.dhsdevelopments.Climaxima.appdata.xml (hunk 2 above)
Comment (lokedhs) [reply com.dhsdevelopments.Climaxima.appdata.xml]: As best as I can tell, the only thing that s missing is the application icon. I will add this shortly.

File: com.dhsdevelopments.Climaxima.json (hunk 3 above)
Comment (lokedhs) [reply com.dhsdevelopments.Climaxima.json]: Thank you. I have made this change.

**Comment by lokedhs** [comment]
bot, build com.dhsdevelopments.Climaxima

**Comment by lokedhs** [comment]
@nedrichards the application is built using Quicklisp, which automatically downloads the dependencies during build. This means I'm getting errors during automated build. Does that mean that I can't submit to flathub?

**Comment by nedrichards** [comment]
@lokedhs For flathub, apps are built in a network isolated container. (this isn't a flatpak limitation, you can absolutely distribute flatpaks built this way yourself on your own infrastructure). You need to specify all the dependencies for reproducibility and source code deliverability. There's a bunch of scripts available for different languages and external packaging systems that help you do this: https://github.com/flatpak/flatpak-builder-tools not currently one for quicklisp but patches are very welcome.

**Comment by lokedhs** [comment]
bot, build com.dhsdevelopments.Climaxima

**Comment by lokedhs** [comment]
bot, build com.dhsdevelopments.Climaxima

**Comment by lokedhs** [comment]
bot, build com.dhsdevelopments.Climaxima

**Comment by lokedhs** [comment]
bot, build com.dhsdevelopments.Climaxima

**Comment by lokedhs** [comment]
bot, build com.dhsdevelopments.Climaxima

**Comment by lokedhs** [comment]
bot, build com.dhsdevelopments.Climaxima

**Comment by nedrichards** [comment]
@lokedhs we have a special way of doing build architecture filtering in an instructional file called flathub.json, that commit will still run everywhere. Also, don't worry about saving resources, they're there to be used and builds are scheduled appropriately. The only reason to remove architectures is if it's impractical or impossible to make it work there.

**Comment by lokedhs** [comment]
bot, build com.dhsdevelopments.Climaxima

**Comment by lokedhs** [comment]
bot, build com.dhsdevelopments.Climaxima

File: com.dhsdevelopments.Climaxima.appdata.xml (hunk 2 above)
Comment (nedrichards) [reply com.dhsdevelopments.Climaxima.appdata.xml]: I ran the validator tool on it and there's more than that required.

``` 
flatpak run org.freedesktop.appstream-glib validate com.dhsdevelopments.Climaxima.appdata.xml
com.dhsdevelopments.Climaxima.appdata.xml: FAILED:
• tag-invalid           : <metadata_license> is not valid [MIT]
• tag-missing           : <content_rating> required [use https://odrs.gnome.org/oars]
• style-invalid         : Not enough <screenshot> tags, minimum is 1
• tag-missing           : <release> required
Validation of files failed
```

Comment (nedrichards) [reply com.dhsdevelopments.Climaxima.appdata.xml]: Huh, and now I see I was reviewing an older commit! Looks good now. I'd add a project_licence tag as well fwiw

**Comment by lokedhs** [comment]
@nedrichards Thank you for the information. I've made the appropriate changes. It is possible to get the application to run on other architectures, but some architectures (especially ARM) requires a different Common Lisp implementation.

I still don't understand Flatpak well enough to be able to construct different build paths depending on architecture.

**Comment by nedrichards** [comment]
bot, build com.dhsdevelopments.Climaxima

**Comment by nedrichards** [comment]
@lokedhs fair enough! it's a bit ungainly I'm afraid, you use `only-arches` or `skip-arches` on the module you want to differ on each architecture. example: https://github.com/flathub/cc.arduino.arduinoide/blob/0242af5a5f7bf59a5200d731158ae1ac15893263/cc.arduino.arduinoide.json#L207

**Comment by lokedhs** [comment]
bot, build com.dhsdevelopments.Climaxima

**Comment by lokedhs** [comment]
Are there any other things that needs to be changed before this is ready for merge?

**Comment by nedrichards** [comment]
bot, build com.dhsdevelopments.Climaxima

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/com.dhsdevelopments.Climaxima

**Pull request closed without merging**